api/tests/unit/features/test_unit_features_permissions.py

```diff
@@ -20,374 +21,492 @@
 )
 from users.models import FFAdminUser, UserPermissionGroup
 
-mock_view = mock.MagicMock()
-mock_request = mock.MagicMock()
 
-feature_permissions = FeaturePermissions()
+def test_organisation_admin_can_list_features(
+    admin_user: FFAdminUser,
+    project: Project,
+) -> None:
+    # Given
+    feature_permissions = FeaturePermissions()
+    mock_view = mock.MagicMock()
+    mock_request = mock.MagicMock()
+    mock_view.kwargs = {}
+    mock_request.data = {}
+    mock_view.action = "list"
+    mock_view.detail = False
+    mock_view.kwargs["project_pk"] = project.id
+    mock_request.user = admin_user
```

Comment:
Since we're now instantiating in each test, I think we can do this: 

```suggestion
    mock_view = mock.MagicMock(kwargs={"project_pk": project.id}, detail=False, action="list")
    mock_request = mock.MagicMock(data={}, user=admin_user)
```

I appreciate that we maybe don't want to make too many changes to the actual tests themselves but since we had to make a change here anyway, I think it makes sense. Happy for you to disagree here though. If we don't want to make the suggested change (and apply similar changes to the rest of the tests) then we should at least remove the duplication since we're assigning a value to `mock_view.kwargs` twice here (in some of the other tests `mock_request.data` is the offender). 

A final option would be to just define 2 fixtures locally in this test module that define the `mock_view` and `mock_request` objects.

Reply:
No this is awesome! I didn't know about putting things in `MagicMock`'s constructors! Have updated!